hardware/esp8266com/esp8266/boards.txt

```diff
@@ -57,6 +58,55 @@ generic.menu.FlashSize.4M=4M
 generic.menu.FlashSize.4M.build.flash_size=4M
 
 ##############################################################
+modwifi.name=Olimex MOD-WIFI-ESP8266(-DEV)
+
+modwifi.upload.tool=esptool
+modwifi.upload.speed=115200
+modwifi.upload.resetmethod=ck
+modwifi.upload.maximum_size=2097152
+modwifi.upload.wait_for_upload_port=true
+modwifi.serial.disableDTR=true
+modwifi.serial.disableRTS=true
+
+modwifi.build.mcu=esp8266
+modwifi.build.f_cpu=80000000L
+modwifi.build.board=MOD_WIFI_ESP8266
+modwifi.build.core=esp8266
+modwifi.build.variant=modwifi
+# Winbond W25Q16 flash
+modwifi.build.flash_mode=qio
+modwifi.build.flash_size=2M
+modwifi.build.flash_freq=40
+
+modwifi.menu.CpuFrequency.80=80 MHz
+modwifi.menu.CpuFrequency.80.build.f_cpu=80000000L
+modwifi.menu.CpuFrequency.160=160 MHz
+modwifi.menu.CpuFrequency.160.build.f_cpu=160000000L
+
+modwifi.menu.UploadSpeed.115200=115200
+modwifi.menu.UploadSpeed.115200.upload.speed=115200
+modwifi.menu.UploadSpeed.9600=9600
+modwifi.menu.UploadSpeed.9600.upload.speed=9600
+modwifi.menu.UploadSpeed.57600=57600
+modwifi.menu.UploadSpeed.57600.upload.speed=57600
+modwifi.menu.UploadSpeed.256000.windows=256000
+modwifi.menu.UploadSpeed.256000.upload.speed=256000
+modwifi.menu.UploadSpeed.230400.linux=230400
+modwifi.menu.UploadSpeed.230400.macosx=230400
+modwifi.menu.UploadSpeed.230400.macosx=230400
+modwifi.menu.UploadSpeed.230400.upload.speed=230400
+modwifi.menu.UploadSpeed.460800.linux=460800
+modwifi.menu.UploadSpeed.460800.macosx=460800
+modwifi.menu.UploadSpeed.460800.upload.speed=460800
+modwifi.menu.UploadSpeed.512000.windows=512000
+modwifi.menu.UploadSpeed.512000.upload.speed=512000
+modwifi.menu.UploadSpeed.921600=921600
+modwifi.menu.UploadSpeed.921600.upload.speed=921600
+
+modwifi.menu.FlashSize.2M=2M
```

Comment:
The Olimex modules that I have to hand are fitted with the 25Q16 *16MB*
Flash, straight from the box - maybe they're something of a variable feast
with Olimex?

Of the five Olimex modules that I have, two had clearly been reworked
around the ESP Chip and one was missing the capacitor that feeds the pcb
antenna - so couldn't possibly work.

Attempts to contact Olimex about the missing capacitor have received no
response.

Other than that, they're breadboard-friendly and have all the right pullups
and pulldowns already fitted (with 'solder-pad jumpers'). They don't have
the uFL antenna connector fitted but the pads are on the board - if you can
find the connectors and have the eyesight to solder them on!

On 6 May 2015 at 07:02, Ivan Grokhotkov <notifications@github.com> wrote:

> In hardware/esp8266com/esp8266/boards.txt
> <https://github.com/esp8266/Arduino/pull/183#discussion_r29734507>:
>
> > +modwifi.menu.UploadSpeed.57600.upload.speed=57600
> > +modwifi.menu.UploadSpeed.256000.windows=256000
> > +modwifi.menu.UploadSpeed.256000.upload.speed=256000
> > +modwifi.menu.UploadSpeed.230400.linux=230400
> > +modwifi.menu.UploadSpeed.230400.macosx=230400
> > +modwifi.menu.UploadSpeed.230400.macosx=230400
> > +modwifi.menu.UploadSpeed.230400.upload.speed=230400
> > +modwifi.menu.UploadSpeed.460800.linux=460800
> > +modwifi.menu.UploadSpeed.460800.macosx=460800
> > +modwifi.menu.UploadSpeed.460800.upload.speed=460800
> > +modwifi.menu.UploadSpeed.512000.windows=512000
> > +modwifi.menu.UploadSpeed.512000.upload.speed=512000
> > +modwifi.menu.UploadSpeed.921600=921600
> > +modwifi.menu.UploadSpeed.921600.upload.speed=921600
> > +
> > +modwifi.menu.FlashSize.2M=2M
>
> If 2M is the only option for flash size, you can remove this. No need to
> have a menu with just a single option.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/esp8266/Arduino/pull/183/files#r29734507>.
>